app/services/widgets.ts

```diff
@@ -388,12 +406,115 @@ export class WidgetsService extends Service {
     return WidgetTesters.filter(tester => {
       return tester.platforms.includes(this.userService.platform.type);
     }).map(tester => {
-      return new WidgetTester(tester.name, tester.url(
-        this.hostsService.streamlabs,
-        this.userService.widgetToken,
-        this.userService.platform.type
-      ));
+      return new WidgetTester(
+        tester.name,
+        tester.url(
+          this.hostsService.streamlabs,
+          this.userService.widgetToken,
+          this.userService.platform.type
+        )
+      );
+    });
+  }
+
+  /**
+   * Save a widget file to the given path
+   * @param path the path to the save the widget file
+   * @param widgetItem the widget to save
+   */
+  async saveWidgetFile(path: string, widgetItem: SceneItem) {
```

Comment:
I would prefer to use serializable parameters for public methods. It will allow to use such methods from child window and expose this method to our public API if we need. So instead `SceneItem` it can be `ISceneItem` or `sceneItemId`. We always can fetch the sceneItem by id.